src/metadataBlocks/infra/repositories/transformers/metadataBlockTransformers.ts

```diff
@@ -51,3 +56,28 @@ const transformPayloadMetadataFieldInfo = (
   }
   return metadataFieldInfo
 }
+
+/**
+ * This method checks if a new metadata field key is already present in the metadata fields.
+ * We need this method since child fields are returned as sub-objects and as root objects in the response payload, so we don't want to replicate this behavior in the model.
+ *
+ * @param {Record<string, MetadataFieldInfo>} [metadataFields] - The current transformed metadata fields
+ * @param {string} newMetadataFieldKey - The new metadata field key to transform
+ *
+ * @returns {boolean}
+ */
```

Comment:
Why did you add a comment to this function when the other transformer functions do not have any? The function is not even an exported function of the package

Reply:
I decided to add a method doc because the behavior of the endpoint (child fields returned both as root objects and sub objects) and the fact that we have to do this filtering is not very intuitive, so I think the doc would help any new dev to understand this better.

Anyway I have removed both the method and the doc, which I don't think is strictly necessary, with the latest changes.